gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/LocalRegion.java

```diff
@@ -14352,6 +14353,12 @@ private long getEntryOverhead(RegionEntry entry){
     Object key = entry.getRawKey();
     if (key != null) {
       entryOverhead += ReflectionSingleObjectSizer.INSTANCE.sizeof(key);
+    }else{
+      // first key.
+      Object firstKey = this.getRegionMap().keySet().iterator().next();
+      if(firstKey != null){
+        entryOverhead += ReflectionSingleObjectSizer.INSTANCE.sizeof(firstKey);
+      }
```

Comment:
Only for row table regions. But meta-table regions, Gateway/AsyncQueue regions etc don't have key as part of values so code here should not assume that. Also being changed for column tables in SNAP-1454.